integration-tests/byoid/integration_test.go

```diff
@@ -202,3 +216,115 @@ func TestFileBasedCredentials(t *testing.T) {
 		},
 	})
 }
+
+// Tests to make sure URL based external credentials work properly.
+func TestURLBasedCredentials(t *testing.T) {
+	//Set up a server to return a token
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			t.Errorf("Unexpected request method, %v is found", r.Method)
+		}
+		w.Write([]byte(oidcToken))
+	}))
+
+	testBYOID(t, config{
+		Type:                           "external_account",
+		Audience:                       oidcAudience,
+		SubjectTokenType:               "urn:ietf:params:oauth:token-type:jwt",
+		TokenURL:                       "https://sts.googleapis.com/v1beta/token",
+		ServiceAccountImpersonationURL: fmt.Sprintf("https://iamcredentials.googleapis.com/v1/%s:generateAccessToken", clientID),
+		CredentialSource: credentialSource{
+			URL: ts.URL,
+		},
+	})
+}
+
+// Tests to make sure AWS based external credentials work properly.
+func TestAWSBasedCredentials(t *testing.T) {
+	data := url.Values{}
+	data.Set("audience", clientID)
+	data.Set("includeEmail", "true")
+
+	client, err := google.DefaultClient(context.Background(), "https://www.googleapis.com/auth/cloud-platform")
+	if err != nil {
+		t.Errof("Failed to create default client: %v", err)
```

Comment:
Would you step through and reevaulate some of these t.Errors, I think some should be t.Fatal such as this one. If there is an error creating the client the rest of the test can't run.

Reply:
Doesn't t.Errorf automatically halt the execution of the test and return an error?